db/StoreArchive.py

```diff
@@ -0,0 +1,55 @@
+import FetchArchive as fetch
+
+from datetime import datetime
+from pymongo import MongoClient
+import os
+import logging
+
+if not os.path.exists('logs'):
+    os.makedirs('logs')
+logging.basicConfig(filename='logs/StoreArchive.log', level=logging.INFO, format='%(asctime)s:%(levelname)s:%(message)s')
+
+db = MongoClient('localhost', 27017).WallStreetDB
+
+
+def buildIndex():
+    '''Gets, cleans and stores data from archiveIndexGenerator().'''
+    form = '%Y-%m-%dT%H:%M:%SZ'
+    for item in fetch.archiveIndexGenerator():
+        # TODO: Add lastModified field.
+        item['_id'] = item['identifier']
+        key = {'_id': item['_id']}
+        if 'date' in item:  # Clean index date. (str -> datetime)
+            item['datetime'] = datetime.strptime(item['date'], form)
+        # Upsert inserts (instead of update) if the item does not exist.
+        db.ArchiveIndex.update_one(key, {'$set': item}, upsert=True)
+
+
+def buildEpisodes():
+    '''
+    Searches local index for empty episodes (documents with no metadata).
+    Scrapes data and updates document.
+    '''
+    num_failed = 0
+    num_suceed = 0
+    empty_episodes = db.ArchiveIndex.find({'metadata': {'$eq': None}})
```

Comment:
Ignores all episodes with metadata already, even if the old metadata followed a different schema. If the schema changes, you have to clear the database, then begin the download process.
On the plus side, you can quit this function and re run it and it will not make requests to already downloaded episodes.